modules/hdr/image_loader_hdr.cpp

```diff
@@ -113,29 +115,49 @@ Error ImageLoaderHDR::load_image(Ref<Image> p_image, Ref<FileAccess> f, BitField
 							}
 						} else {
 							// Dump
+							f->get_buffer(temp_read_ptr, count);
 							for (int z = 0; z < count; ++z) {
-								ptr[(j * width + i++) * 4 + k] = f->get_8();
+								ptr[(j * width + i++) * 4 + k] = temp_read_ptr[z];
 							}
 						}
 					}
 				}
 			}
 		}
 
+		const bool force_linear = p_flags & FLAG_FORCE_LINEAR;
+
 		//convert
 		for (int i = 0; i < width * height; i++) {
-			float exp = pow(2.0f, ptr[3] - 128.0f);
+			int e = ptr[3] - 128;
+
+			if (force_linear || (e < -15 || e > 15)) {
+				float exp = pow(2.0f, e);
+				Color c(ptr[0] * exp / 255.0, ptr[1] * exp / 255.0, ptr[2] * exp / 255.0);
 
-			Color c(
-					ptr[0] * exp / 255.0,
-					ptr[1] * exp / 255.0,
-					ptr[2] * exp / 255.0);
+				if (force_linear) {
+					c = c.srgb_to_linear();
+				}
 
-			if (p_flags & FLAG_FORCE_LINEAR) {
-				c = c.srgb_to_linear();
+				*(uint32_t *)ptr = c.to_rgbe9995();
+			} else {
+				// https://github.com/george-steel/rgbe-rs/blob/main/src/types.rs
+				// MIT License
```

Comment:
Same here. Copyright doesn't apply to 2 lines of code. We should point to the source though (a stable branch, not the main branch which may change)

Reply:
This one doesn't have any other branch besides 'main', so I'm unsure of what to do here

Reply:
You can copy a permalink to the section of code. For example: https://github.com/george-steel/rgbe-rs/blob/e7cc33b7f42b4eb3272c166dac75385e48687c92/src/types.rs#L61-L80

Reply:
To do that, you select the lines of code you are interested in, then click on the three dot icon that appears beside the line numbers and select "copy permalink"

![image](https://github.com/user-attachments/assets/e062c49e-429a-4cd5-8650-d65a9932a83f)